### PR TITLE
feature(crc): upload to district magistrate

### PR DESCRIPTION
**Pull request by Rikkiabb**
# \<Description\>
Uploads signed contract and data to sýslumenn starfskerfi when both parents have signed contract.
Uses application template api to call action to upload data when application is submitted. 
Uses already created syslumenn service to call starfskerfi.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts
Comment (Rikkiabb) [inline libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts:36]: I removed `/dev` from this url, since we are reusing the login funcitonality. 
This means we will have to update the value in helm. Both I think we would have to do that in the end anyways.

File: libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts
Comment (Rikkiabb) [inline libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts:40]: FYI: Since the login response returns a value called `accessToken` which I have to use to upload data then I changed the variable name of the variable formerly known as `accessToken` in this scope to `id`

File: libs/application/template-api-modules/src/lib/modules/templates/children-residence-change/children-residence-change.service.ts
Comment (VilliGunn) [inline libs/application/template-api-modules/src/lib/modules/templates/children-residence-change/children-residence-change.service.ts:61]: This is just nitpicking but we could deconstruct like this:
```
const { answers, externalData } = application
const { parentNationalRegistry } = externalData
```

And save a few characters 😄

File: libs/application/templates/children-residence-change/src/lib/ChildrenResidenceChangeTemplate.ts
Comment (VilliGunn) [inline libs/application/templates/children-residence-change/src/lib/ChildrenResidenceChangeTemplate.ts:114]: Is it not possible to do `id: Roles.ParentA || Roles.ParentB` here?

File: libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts
Comment (Ingimarsson) [inline libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts:60]: Should it not be like this?

```suggestion
    const url = `${this.clientConfig.url}/v1/SyslMottakaGogn`
```

File: libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts
Comment (Ingimarsson) [reply libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts:36]: Nice. Can you make the changes in the helm repo? I guess you'll need to change the host in `portal` and add all the variables in `application-system`

**Review by Ingimarsson — CHANGES_REQUESTED** [review]

File: libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts
Comment (Rikkiabb) [reply libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts:60]: No but it should be ☹️ But as you can see here: https://api.syslumenn.is/dev/swagger/index.html#/ then for whatever reason then this endpoint has and extra `/api/` in it's path.

Comment (Ingimarsson) [reply libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts:60]: Interesting :thinking:

File: libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts
Comment (Rikkiabb) [reply libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts:36]: Of course 😄  I was thinking of creating a single pr in helm where I update it after this one get's merged. How does that sound? :)

**Review by Ingimarsson — APPROVED** [review]
LGTM (syslumenn client)

File: libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts
Comment (Ingimarsson) [reply libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts:36]: Great!

**Review by dabbeg — APPROVED** [review]
Looks good!

**Review by VilliGunn — APPROVED** [review]
💯

File: apps/application-system/api/src/environments/environment.ts
Comment (baering) [inline apps/application-system/api/src/environments/environment.ts:37]: Same as above.

File: libs/api/domains/syslumenn/src/lib/models/dataUpload.ts
Comment (baering) [inline libs/api/domains/syslumenn/src/lib/models/dataUpload.ts:7]: Is this intentional? Array of one `Person` and one `Attachment`.

**Review by baering — COMMENTED** [review]
Nice, LGTM 👍   Just have some minor comments/questions before approval.

File: apps/application-system/api/src/environments/environment.prod.ts
Comment (Rikkiabb) [reply apps/application-system/api/src/environments/environment.prod.ts:32]: This should be used for all api's from syslumenn within the application system. 

This is a valid point since this is pretty specific and currently I don't know how many endpoints we will use from syslumenn.

File: libs/api/domains/syslumenn/src/lib/models/dataUpload.ts
Comment (Rikkiabb) [reply libs/api/domains/syslumenn/src/lib/models/dataUpload.ts:7]: 🤦  nice catch 👏

**Review by baering — APPROVED** [review]

**Pull request merged**
